### PR TITLE
docs: Improves `Table` documentation

### PR DESCRIPTION
**Pull request by ling1726**
- Cleanup docstrings
- Adds main component description
- Adds descriptions to relevant examples
- Adds subcomponent props
- Reorders examples

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.975 kB`<br />`17.687 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `190.411 kB`<br />`53.077 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `33.48 kB`<br />`11.037 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/56a61fc731fa4ca6440eb78629c1e38b26154ffa'>56a61fc731fa4ca6440eb78629c1e38b26154ffa</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/Avatar-mount_base.html">1697</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/Avatar-mount.html">1648</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/Button-mount_base.html">1174</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/Button-mount.html">1149</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProvider-mount_base.html">2006</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProvider-mount.html">1959</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProviderWithTheme-mount_base.html">704</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProviderWithTheme-mount.html">744</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">687</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">684</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">719</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">712</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/MakeStyles-mount_base.html">2324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/MakeStyles-mount.html">2344</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/SpinButton-mount_base.html">3305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25787/perf-test-react-components/SpinButton-mount.html">3318</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 2 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-tool-fluent/index.html?organization=uifabric&project=fabricpublic&prId=25787&commitId=0b975170b504e81fcb1b90ec862a172673a27837&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1749 screenshots unchanged, 0 screenshots with different dimensions and 2 screenshots with visible difference.


  <details><summary><b>unknown</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.size+active+badge.normal.chromium.png|17| Changed|
|Avatar Converged.size+active+ring-shadow.normal.chromium.png|16| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-table/stories/Table/CellActions.stories.tsx [hunk 1]
```diff
@@ -112,3 +112,16 @@ export const CellActions = () => {
     </Table>
   );
 };
+
+CellActions.parameters = {
+  docs: {
+    description: {
+      story: [
+        '`TableCellActions` is a container that is visible when the `TableRow`is hovered or focus lies within the',
```
Comment (miroslavstastny) [inline packages/react-components/react-table/stories/Table/CellActions.stories.tsx:120]: Fix the sentence

File: packages/react-components/react-table/stories/Table/CellNavigation.stories.tsx [hunk 2]
```diff
@@ -108,3 +108,15 @@ export const CellNavigation = () => {
...
   );
 };
+
+CellNavigation.parameters = {
+  docs: {
+    description: {
+      story: [
+        'The `Table` primitive components do not support keyboard navigation. This should be added by users.',
+        'Cell navigation can be achieved simply using the `useArrowNavigationGroup` utility provided by the Library.',
+        'user state.',
```
Comment (miroslavstastny) [inline packages/react-components/react-table/stories/Table/CellNavigation.stories.tsx:118]:
```suggestion
```

File: packages/react-components/react-table/stories/Table/CellNavigation.stories.tsx [hunk 3]
```diff
@@ -108,3 +108,15 @@ export const CellNavigation = () => {
     </Table>
   );
 };
+
+CellNavigation.parameters = {
```
Comment (miroslavstastny) [inline packages/react-components/react-table/stories/Table/CellNavigation.stories.tsx:112]: not related to the PR, but the fact that storybook shows only the exported component source code makes the examples really hard to read - I do not know what the `columns` and `items` are, where you import `useArrowNavigationGroup` from...  :/

File: packages/react-components/react-table/stories/Table/MultipleSelect.stories.tsx [hunk 4]
```diff
@@ -184,3 +184,15 @@ export const MultipleSelect = () => {
     </Table>
   );
 };
+
```
Comment (miroslavstastny) [inline packages/react-components/react-table/stories/Table/MultipleSelect.stories.tsx:187]: Please double check keyboard navigation - I can select the row checkbox by both Enter and Space but Space scrolls.
For the uber checkbox for all lines I can only select it using Space (without scroll) cannot select it using Enter.

**Review by miroslavstastny — APPROVED** [review]
lgtm

File: packages/react-components/react-table/stories/Table/CellNavigation.stories.tsx (hunk 3 above)
Comment (ling1726) [reply packages/react-components/react-table/stories/Table/CellNavigation.stories.tsx:112]: yeah we've had an open issue on it for a while #20046

File: packages/react-components/react-table/stories/Table/CellActions.stories.tsx [hunk 5]
```diff
@@ -112,3 +112,16 @@ export const CellActions = () => {
     </Table>
   );
 };
+
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-components/react-table/stories/Table/CellActions.stories.tsx:115]: Submitted a bug for this #25789

File: packages/react-components/react-table/stories/Table/MultipleSelect.stories.tsx (hunk 4 above)
Comment (ling1726) [reply packages/react-components/react-table/stories/Table/MultipleSelect.stories.tsx:187]: Good point, I've made it behave consistently to uber checkbox and added `preventDefault` to stop the scrolling

**Pull request merged**